### DISCUSSION TITLE
2022梅竹黑客松台積電組 比賽經驗分享

### DISCUSSION THREAD

**Discussion by gnsJhenJie** [1101-GS3073-C/Python-Community-GS3073-C#482] [Show and tell]
上週六日參加了今年十週年的梅竹黑客松，梅竹黑客松會分成由各企業出題且需要實做出成品的**黑客組**和只需提出構想的**創客組**(這名字每年可能不太一樣)，比賽主要是希望參賽者在為期兩天的賽程中不間斷的腦力激盪及coding(不過想要偷跑當然也是可以的，我們這隊賽前一週就有討論出要做的方向，並且寫了程式大致的架構XD)。另外，雖然黑客組比賽需要coding的，但並不是資工系才能參加(像我們隊的隊員就只有兩個資工的，其他分別是資管、財務管理和機械)。  
- 隊員們
<img width="527" alt="Screenshot 2022-10-30 at 02 17 20" src="https://user-images.githubusercontent.com/32983347/198846948-2bcfc87e-1b92-432d-9fd9-12dbd193175e.png">

我這隊選擇的黑客組中的台積電組，主題是希望我們做出能辨識工地工人有無戴安全帽的AI模型，並且在雲端平台建立應用程式。台積在這場比賽派出超多mentor來讓我們技術支援(?)，大概每1.5個參賽者就有一個mentor，很多看起來看起來都是資深的工程師(AI, Big data, , VR, 網頁的team都有)，還提供GCP 40小時的兩張V100 GPU讓我們使用(價值大約一萬台幣，不過我們這隊主要還是用實驗室的GPU訓練)，可見他們對這次比賽的重視。
- 比賽現場，一組一桌，也可以挑自己喜歡的位置coding
![IMG_9632](https://user-images.githubusercontent.com/32983347/198847398-85f61450-878a-4005-9ad6-17299a2fdfc7.jpg)

我們這隊的作品是一個工地安全管理系統，透過yolov7訓練的模型結合樹莓派監控工地是否有人沒戴安全帽，並且實作了網頁管理介面及聊天機器人即時通知工地管理者(至於技術細節可能之後再發一篇吧?)，最後還不小心拿了第二名XD。  
- 網頁本人 (我們用的AWS要錢，比賽結束就趕緊關掉，忘記先截幾張圖，只好從簡報找圖...)
![Screenshot 2022-10-30 at 02 21 20](https://user-images.githubusercontent.com/32983347/198847077-60aed6cc-d6c7-431c-a0fd-0058eefdd47d.png)
- 樹莓派本人 (那顆喇叭還會出中英台三語提醒，台語部分是用成大的AI生成的(單純覺得很酷才放上去的功能XD)，另外感謝 @benctw 教授借webcam)
![圖片 1](https://user-images.githubusercontent.com/32983347/198847169-77104229-5432-4f9d-ab0f-a392a0341e1e.jpg)

除了認真coding外，現場還有許多遊樂設施，和不間斷供應的零食飲料，像是下圖的球池我就很喜歡，晚上我們就睡在上面(說到睡覺，我三天大概睡六小時)，午餐晚餐的部分都是吃到飽，大概都有二十道菜可以拿。

- 好睡的球池
![球池](https://user-images.githubusercontent.com/32983347/198846405-fb33551e-e59e-486e-ae3b-eec39e7cb58d.GIF)
- 晚餐
<img src="https://user-images.githubusercontent.com/32983347/198846550-6b686400-733d-4e72-b8b9-a9a517d592d9.JPG" data-canonical-src="https://user-images.githubusercontent.com/32983347/198846550-6b686400-733d-4e72-b8b9-a9a517d592d9.JPG" width="300"  />

關於比賽內容大概就醬，總之還蠻推薦大家來比比看的(去年的黑客松我是參加創客組也有拿獎，有用心準備我認為拿獎應該不難)，如果有什麼問題也可以在底下留言，我盡量回答~

**Comment by lx3211** [comment]
想請問財管的同學大致是負責什麼樣的工作呢

**Reply by gnsJhenJie** [reply]
他主要是想要做哪些功能、做簡報和報告。另一個會找他的原因是我們報名企業的第二志願是選中國信託，不過後來第一志願就上了，所以沒什麼用到他財管方面的技能。

**Reply by lx3211** [reply]
真是特別的經歷，我發現你們組員學校幾乎都不一樣，是找以前的朋友組隊嗎？還是為了比賽招募的呢？

**Reply by gnsJhenJie** [reply]
都是以前高中同學, 之前就有一起參加過其他比賽了, 我覺得找彼此熟悉的隊友大部分時候會比找一群很強但彼此不熟悉的好。

**Comment by Chenszuhong** [comment]
我覺得這比賽超帥的，希望自己能力變強之後可以有有機會參加

**Comment by xCk27x** [comment]
想請問在資工方面
需要甚麼能力再來參加這個比賽比較好呢?

**Reply by gnsJhenJie** [reply]
以參加這類黑客松比賽來說的話，我覺得以下三個是比較重要的
- Git: 基本上很多人同時寫code的話git很重要，另外在程式碼上的分配也很重要，如果很多人都在編輯相互依賴的function時，很容易需要處理conflict
- Machine Learning: 現在很多的比賽題目都會跟ML相關，如果題目沒有提到ML，其他對手也很可能會有用到ML，這部分雖然不用自己一個layer一個layer堆出模型，可以套別人的模型就好，但還是需要一些相關知識，像是你對訓練模型的參數或data augmentation做了哪些調整、為什麼要這樣做，否則報告的時候就只能說「我套了XXX模型」會顯得在ML方面很虛。
- 一個能展現成果的工具: 通常會需要一個網頁、Line Bot或App之類的視覺化工具來展現成果，這是讓評審們了解到你的演算法是真的能實際應用的一個好的方法。

另外，這些能力不一定要每個人都有，整個團隊能有這些就好了(也可以多找找不同的比賽，每個比賽需要的能力也會有些差異)。

**Comment by williamyen0623** [comment]
助教也太帥，什麼都會，每次去上課問問題，都能秒解
以後有機會我一定也要去參加黑客松！

**Comment by NcuJerryKao** [comment]
助教好強 不肖新拿第二名 去年參加也有得獎
三天只睡六小時還能coding 太扯的啦 我的話 要是沒睡飽 什麼都打不出來

**Comment by BingRong1217** [comment]
好酷好有趣的比賽，感覺可以學習到很多東西

**Comment by limitx0** [comment]
助教太厲害了，不小心就第二名
梅竹黑客松的比賽內容和場地設備好酷
